packages/ai-jsx/src/lib/anthropic.tsx

```diff
@@ -0,0 +1,178 @@
+} from '../core/completion.js';
+import { AIJSXError, ErrorCode } from '../core/errors.js';
+
+export const anthropicClientContext = AI.createContext<Anthropic>(
+  new Anthropic({
+    apiKey: getEnvVar('ANTHROPIC_API_KEY', false),
+  })
+);
+
+type ValidCompletionModel = never;
+/**
+ * The set of valid Claude models.
+ *
+ * @see https://docs.anthropic.com/claude/reference/complete_post.
+ */
+type ValidChatModel =
+  | 'claude-1'
+  | 'claude-1-100k'
+  | 'claude-instant-1'
+  | 'claude-instant-1-100k'
+  | 'claude-1.3'
+  | 'claude-1.3-100k'
+  | 'claude-1.2'
+  | 'claude-1.0'
+  | 'claude-instant-1.1'
+  | 'claude-instant-1.1-100k'
+  | 'claude-instant-1.0';
+
+type AnthropicModelChoices = ChatOrCompletionModelOrBoth<ValidChatModel, ValidCompletionModel>;
+
+/**
+ * An AI.JSX component that invokes an Anthropic Large Language Model.
+ * @param children The children to render.
+ * @param chatModel The chat model to use.
+ * @param completionModel The completion model to use.
+ * @param client The OpenAI client.
+ */
+export function OpenAI({
+  children,
+  chatModel,
+  completionModel,
+  client,
+  ...defaults
+}: { children: Node; client?: Anthropic } & AnthropicModelChoices & ModelProps) {
+  let result = children;
+
+  if (client) {
+    result = <anthropicClientContext.Provider value={client}>{children}</anthropicClientContext.Provider>;
+  }
+
+  if (chatModel) {
+    result = (
+      <ChatProvider component={AnthropicChatModel} {...defaults} model={chatModel}>
+        {result}
+      </ChatProvider>
+    );
+  }
+
+  // TS is correct that this should never happen, but we'll check for it anyway.
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (completionModel) {
+    throw new AIJSXError(
+      'Completion models are not supported by Anthropic',
+      ErrorCode.AnthropicDoesNotSupportCompletionModels,
+      'user'
+    );
+  }
+
+  return result;
+}
+
+interface AnthropicChatModelProps extends ModelPropsWithChildren {
+  model: ValidChatModel;
+}
+export async function* AnthropicChatModel(
+  props: AnthropicChatModelProps,
+  { render, getContext, logger }: AI.ComponentContext
+): AI.RenderableStream {
+  const messageElements = await render(props.children, {
+    stop: (e) =>
+      e.tag == SystemMessage ||
+      e.tag == UserMessage ||
+      e.tag == AssistantMessage ||
+      e.tag == FunctionCall ||
+      e.tag == FunctionResponse,
+  });
+  yield AI.AppendOnlyStream;
+  const messages = await Promise.all(
+    messageElements.filter(AI.isElement).map(async (message) => {
+      switch (message.tag) {
+        case UserMessage:
+          return `${Anthropic.HUMAN_PROMPT}: ${await render(message)}`;
+        case AssistantMessage:
+          return `${Anthropic.AI_PROMPT}: ${await render(message)}`;
+        case SystemMessage:
+          throw new AIJSXError(
+            'Anthropic models do not support SystemMessage. Change your user message to instruct the model what to do.',
```

Comment:
should we just jam this in as a UserMessage then?

Reply:
https://github.com/fixie-ai/ai-jsx/pull/165#discussion_r1246036009